### PR TITLE
Loosen type condition in evaluation functions

### PR DESCRIPTION
**Pull request by okuta**
These fcuntions accept an array of numpy.float16, numpy.float32 and numpy.float64.
See also #555

### CODE REVIEW

**Comment by delta2323** [comment]
I have also fixed `Accuracy` in #1162.

File: chainer/functions/evaluation/binary_accuracy.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def forward(self, inputs):
         t = t.ravel()
         c = (y >= 0)
         count = xp.maximum(1, (t != self.ignore_label).sum())
-        return xp.asarray((c == t).sum(dtype='f') / count, dtype='f'),
+        return xp.asarray((c == t).sum(dtype=y.dtype) / count, dtype=y.dtype),
```
Comment (delta2323) [inline chainer/functions/evaluation/binary_accuracy.py]: Do we have to change `dtype` twice?

Comment (okuta) [inline chainer/functions/evaluation/binary_accuracy.py]: Yes. But I have better idea. I will fix it.

File: chainer/functions/evaluation/binary_accuracy.py [hunk 2]
```diff
@@ -27,7 +29,7 @@ def forward(self, inputs):
         t = t.ravel()
         c = (y >= 0)
         count = xp.maximum(1, (t != self.ignore_label).sum())
-        return xp.asarray((c == t).sum(dtype='f') / count, dtype='f'),
+        return xp.asarray((c == t).sum() / count, dtype=y.dtype),
```
Comment (delta2323) [inline chainer/functions/evaluation/binary_accuracy.py]: I think it is a division of an integer by an integer. So it causes a rounding to the integer, doesn't it?

Comment (okuta) [inline chainer/functions/evaluation/binary_accuracy.py]: It uses Python3 mode.
https://github.com/pfnet/chainer/pull/1165/files/91b37a010ae0ff64115e1fbc54b427f292b9920c#diff-36d8cd578e4268189b044f7fad32c5e3R1

Comment (delta2323) [inline chainer/functions/evaluation/binary_accuracy.py]: Sorry, I missed it.

**Comment by delta2323** [comment]
LGTM 👍

**Pull request merged**